src/main/scala/mesosphere/marathon/core/launchqueue/impl/ReviveOffersActor.scala

```diff
@@ -0,0 +1,122 @@
+package mesosphere.marathon
+package core.launchqueue.impl
+
+import akka.actor.{Actor, Props, Stash, Status}
+import akka.event.{EventStream, LoggingReceive}
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.event.InstanceChanged
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.launchqueue.ReviveOffersConfig
+import mesosphere.marathon.core.task.tracker.InstanceTracker
+import mesosphere.marathon.metrics.{Counter, Metrics}
+
+import scala.collection.immutable.HashSet
+
+class ReviveOffersActor(
+    metrics: Metrics,
+    conf: ReviveOffersConfig,
+    eventStream: EventStream,
+    instanceTracker: InstanceTracker,
+    driverHolder: MarathonSchedulerDriverHolder) extends Actor with Stash with StrictLogging {
+
+  private[this] val reviveCountMetric: Counter = metrics.counter("mesos.calls.revive")
+  private[this] val suppressCountMetric: Counter = metrics.counter("mesos.calls.suppress")
+
+  override def preStart(): Unit = {
+    super.preStart()
+
+    import akka.pattern.pipe
+    import context.dispatcher
+    instanceTracker.instancesBySpec().pipeTo(self)
+
+    eventStream.subscribe(self, classOf[InstanceChanged])
```

Comment:
Let's use the instance tracker subscription mechanism to ensure that we have consistent replication of state.